plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Diagnostics.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE TupleSections         #-}
+module Ide.Plugin.Cabal.Diagnostics
+( errorDiagnostic
+, warningDiagnostic
+  -- * Re-exports
+, FileDiagnostic
+, Diagnostic(..)
+)
+where
+
+import qualified Data.Text              as T
+import           Development.IDE        (FileDiagnostic,
+                                         ShowDiagnostic (ShowDiag))
+import           Distribution.Fields    (showPError, showPWarning)
+import qualified Ide.Plugin.Cabal.Parse as Lib
+import           Language.LSP.Types     (Diagnostic (..),
+                                         DiagnosticSeverity (..),
+                                         DiagnosticSource, NormalizedFilePath,
+                                         Position (Position), Range (Range),
+                                         fromNormalizedFilePath)
+
+-- | Produce a diagnostic from a Cabal parser error
+errorDiagnostic :: NormalizedFilePath -> Lib.PError -> FileDiagnostic
+errorDiagnostic fp err@(Lib.PError pos _) =
+  mkDiag fp (T.pack "parsing") DsError (toBeginningOfNextLine pos) msg
+  where
+    msg = T.pack $ showPError (fromNormalizedFilePath fp) err
+
+-- | Produce a diagnostic from a Cabal parser warning
+warningDiagnostic :: NormalizedFilePath -> Lib.PWarning -> FileDiagnostic
+warningDiagnostic fp warning@(Lib.PWarning _ pos _) =
+  mkDiag fp (T.pack "parsing") DsWarning (toBeginningOfNextLine pos) msg
+  where
+    msg = T.pack $ showPWarning (fromNormalizedFilePath fp) warning
```

Comment:
Can we simplify the warning message? I'm seeing
```
[parsing] [E] /home/kokobd/work/github.com/haskell/haskell-language-server/plugins/hls-cabal-plugin/hls-cabal-plugin.cabal:9:25:
unexpected Unknown SPDX license identifier: 'MIT2'
```
But I believe the file position is not needed, as the message already shows at that location.

Reply:
They are pretty much unstructured, I am not entirely sure that's possible :( but we can give it a try, nevertheless.

Reply:
Resolved for a follow-up PR, so this is no longer blocks the other hls-cabal-plugin PRs.